packages/portfolio-contract/src/resolver/resolver.exo.ts

```diff
@@ -0,0 +1,253 @@
+  CCTPSettlementOfferArgs,
+} from './types.js';
+import { CCTPTransactionDetailsShape } from './types.js';
+
+const trace = makeTracer('Resolver');
+
+/**
+ * Prepare the CCTP Transaction Resolver.
+ *
+ * This is an orchestration component that can be used independently
+ * of portfolio logic to handle CCTP transaction confirmations.
+ *
+ * @param zone - Durable storage zone
+ * @param vowTools - Vow tools for creating promises
+ * @returns A function to create resolver instances with separate facets
+ */
+export const prepareCCTPResolver = (zone: Zone, vowTools: VowTools) => {
+  const resolverZone = zone.subZone('CCTPResolver');
+  const cctpTransactionRegistry = resolverZone.mapStore<
+    CCTPTransactionKey,
+    CCTPTransactionEntry
+  >('cctpTransactionRegistry');
+
+  const ClientFacetI = M.interface('CCTPResolverClient', {
+    registerCCTPTransaction: M.call(M.string(), M.bigint()).returns(VowShape),
+  });
+
+  const ServiceFacetI = M.interface('CCTPResolverService', {
+    settleCCTPTransaction: M.call(
+      M.string(),
+      M.string(),
+      M.bigint(),
+      M.string(),
+    ).returns(),
+  });
+
+  const DebugFacetI = M.interface('CCTPResolverDebug', {
+    getPendingTransactions: M.call().returns(M.arrayOf(M.string())),
+  });
+
+  return resolverZone.exoClassKit(
+    'CCTPResolver',
+    {
+      client: ClientFacetI,
+      service: ServiceFacetI,
+      debug: DebugFacetI,
+    },
+    () => ({}),
+    {
+      client: {
+        registerCCTPTransaction(destinationAddress: AccountId, amount: bigint) {
+          const key = `${destinationAddress}:${amount}` as CCTPTransactionKey;
+
+          if (cctpTransactionRegistry.has(key)) {
+            trace(`CCTP transaction already registered: ${key}`);
+            return cctpTransactionRegistry.get(key).vowKit.vow;
+          }
+
+          const vowKit = vowTools.makeVowKit<void>();
+
+          cctpTransactionRegistry.init(
+            key,
+            harden({
+              destinationAddress,
+              amountValue: amount,
+              vowKit,
+            }),
+          );
+
+          trace(`Registered pending CCTP transaction: ${key}`);
+          return vowKit.vow;
+        },
+      },
+      service: {
+        settleCCTPTransaction(
+          chainId: CaipChainId,
+          remoteAddress: `0x${string}`,
+          amountValue: bigint,
+          status: 'confirmed' | 'failed' | 'pending',
+        ) {
+          const destinationAddress: AccountId = `${chainId}:${remoteAddress}`;
+          const key =
+            `${destinationAddress}:${amountValue}` as CCTPTransactionKey;
+
+          if (!cctpTransactionRegistry.has(key)) {
+            trace('No pending CCTP transaction found for key:', key);
+            trace('Available keys in registry:', [
+              ...cctpTransactionRegistry.keys(),
+            ]);
+            throw Error(
+              `No pending CCTP transaction found matching provided details: ${key}`,
+            );
```

Comment:
We should be using `@endo/errors` thrughout the contract.

```suggestion
            throw assert.error(
              X`No pending CCTP transaction found matching provided details: ${q(key)}`,
            );
```

That's assuming the key is not sensitive; that is: it's ok to share with whoever gets this error message. I think that's the case. Otherwise:

```suggestion
            throw assert.error(
              X`No pending CCTP transaction found matching provided details: ${key}`,
            );
```